src/client/dfuse/pil4dfs/hook.c

```diff
@@ -97,12 +97,29 @@ static char     *path_libpthread;
 #define MAX_MAP_SIZE	(512*1024)
 #define MAP_SIZE_LIMIT	(16*1024*1024)
 
+/*
+ * get_string_pos - Determine the start and end positions of a string.
+ */
+static void
+get_string_pos(char *buf, char **start, char **end)
+{
+	*start = buf;
+	while (**start != ' ') {
```

Comment:
Going backwards through a string just seems dangerous, how do you know this won't drop off the front of the string?

Reply:
ok. I will add boundary check.